app/src/parser.c

```diff
@@ -193,6 +193,37 @@ parser_error_t parser_printArgument(const flow_argument_list_t *v,
     return parser_ok;
 }
 
+
+parser_error_t parser_printArgumentOptionalDelegatorID(const flow_argument_list_t *v,
+                                               uint8_t argIndex, char *expectedType, jsmntype_t jsonType,
+                                               char *outVal, uint16_t outValLen,
+                                               uint8_t pageIdx, uint8_t *pageCount) {
+    MEMZERO(outVal, outValLen);
+
+    if (argIndex >= v->argCount) {
+        return parser_unexpected_number_items;
+    }
+
+    *pageCount = 1;
+
+    parsed_json_t parsedJson = {false};
+    CHECK_PARSER_ERR(json_parse(&parsedJson, (char *) v->argCtx[argIndex].buffer, v->argCtx[argIndex].bufferLen));
+    uint16_t valueTokenIndex;
+    CHECK_PARSER_ERR(json_matchOptionalKeyValue(&parsedJson, 0, expectedType, jsonType, &valueTokenIndex))
+    if (valueTokenIndex == JSON_MATCH_VALUE_IDX_NONE) {
+        if (outValLen < 5) {
```

Comment:
hmm, isn't there some strncpy helper? :thinking:

Reply:
Yes, there is strncpy_s funtion. I'll change it.